### PR TITLE
Convert NetworkStatus

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Convert `NetworkStatus` to Swift

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6649,7 +6642,7 @@
 						CreatedOnToolsVersion = 8.0;
 						DevelopmentTeam = W5KEQBF9B5;
 						LastSwiftMigration = 1000;
-						ProvisioningStyle = Manual;
+						ProvisioningStyle = Automatic;
```
Comment (typfel) [inline Wire-iOS.xcodeproj/project.pbxproj:6645]: Unrelated changes

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -8530,13 +8522,21 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = F1B24239219B2C9B00F7035C /* Share-Extension-Debug.xcconfig */;
 			buildSettings = {
+				CODE_SIGN_IDENTITY = "iPhone Developer";
```
Comment (typfel) [inline Wire-iOS.xcodeproj/project.pbxproj:8525]: Unrelated changes

File: Wire-iOS Share Extension/ShareExtensionViewController.swift [hunk 3]
```diff
@@ -118,7 +116,6 @@ final class ShareExtensionViewController: SLComposeServiceViewController {
         let activity = ExtensionActivity(attachments: extensionContext?.attachments.sorted)
         sharingSession?.analyticsEventPersistence.add(activity.openedEvent())
         extensionActivity = activity
-        NetworkStatus.add(netObserver)
```
Comment (typfel) [inline Wire-iOS Share Extension/ShareExtensionViewController.swift:121]: This was starting the reachability observing if wasn't already running. Do we start it somewhere else now?

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 2 above)
Comment (billypchan) [reply Wire-iOS.xcodeproj/project.pbxproj]: fixed

File: Wire-iOS Share Extension/ShareExtensionViewController.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:121]: This line was create an observation `ShareExtensionNetworkObserver` which post a notification when observes a `NetworkStatusNotificationName` notification. In this PR I change to directly observing `Notification.Name.NetworkStatus` instead.

Comment (typfel) [reply Wire-iOS Share Extension/ShareExtensionViewController.swift:121]: Yep, that I understood, I was concerned that we didn't create an instance of `NetworkStatus` anymore, which is actually posting the notifications. But I see now that we access `NetworkStatus.shared.reachabillity` from the `SendingProgressViewController`, which should create an instance.
